crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -816,15 +821,15 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
 
         let mut header = None;
         if let Some(num) = self.block_indices.get_canonical_block_number(hash) {
-            header = self.externals.database().header_by_number(num)?;
+            header = self.externals.database().provider()?.header_by_number(num)?;
```

Comment:
This is making it more obvious / explicit now that we eg could be reusing the TX here